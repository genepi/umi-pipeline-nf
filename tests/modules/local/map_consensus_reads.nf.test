nextflow_process {

    name "Test Process MAP_CONSENSUS"
    script "modules/local/umi_polishing/map_consensus.nf"
    process "MAP_CONSENSUS"

    test("Maps consensus reads and should run with success") {

        when {
            process {
                """
                input[0] = Channel.from(
                    ["barcode03", "lpa2645", "$baseDir/tests/input/mapping/barcode03/consensus/masked_consensus.fastq"], 
                    ["barcode04", "lpa2645", "$baseDir/tests/input/mapping/barcode04/consensus/masked_consensus.fastq"], 
                    ["barcode05", "lpa2645", "$baseDir/tests/input/mapping/barcode05/consensus/masked_consensus.fastq"], 
                    )
                input[1] = "consensus"
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.sort).match()
        }

    }

    test("Maps final consensus reads and should run with success") {

        when {
            process {
                """
                input[0] = Channel.from(
                    ["barcode03", "lpa2645", "$baseDir/tests/input/mapping/barcode03/final/final.fastq"], 
                    ["barcode04", "lpa2645", "$baseDir/tests/input/mapping/barcode04/final/final.fastq"], 
                    ["barcode05", "lpa2645", "$baseDir/tests/input/mapping/barcode05/final/final.fastq"], 
                    )
                input[1] = "final"
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.sort).match()
        }

    }

}
