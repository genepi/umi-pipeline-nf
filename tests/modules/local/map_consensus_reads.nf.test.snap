{
    "Maps consensus reads and should run with success": {
        "content": null,
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-26T11:10:59.114637"
    },
    "Maps final consensus reads and should run with success": {
        "content": null,
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-26T11:11:03.840204"
    }
}